src/profilers.cpp

```diff
@@ -262,17 +269,21 @@ namespace aperf {
             nanosleep(&ts, NULL);
         }
 
+        command_status = waitpid(pid, &status, WNOHANG); //check if command to be profiled is still executing
+       
+
       }
       //return code for metric exec
-      std::exit(code_metric_exec);
-    }
+      return code_metric_exec;
+    });
 
+    int result_metric_profiler = metric_profiler.get();
```

Comment:
This hangs the start method until the profiled command finishes executing. You should move it inside ```this->process```.

Reply:
Done :+1: